### ISSUE TITLE
Priority: Target for IEC materials pulling from wrong source 

### ISSUE THREAD

**Issue by eefronson** [BlueCodeSystems/nacmis#281]
<img width="303" alt="screen shot 2018-07-11 at 8 30 55 am" src="https://user-images.githubusercontent.com/38062748/42554365-e0b2e448-84e4-11e8-9a1f-9ff2f1bb9fbe.png">

The target population for IEC materials is pulling from the wrong data source, it should not be pulling from organization type but TARGET POPULATION for the materials themselves. I tried to change the source in superset but it is not an option to pull from.

**Comment by AndreLesa** [comment]
@zephyrfalcon  - This situation is similar to the mobile workers type. We need to pull from data_entry_iecmaterial2_target_audience.

This is because data_entry_iecmaterial2_target_audience is the actual bridging table and the relationship is a Many2Many.

**Comment by zephyrfalcon** [comment]
@AndreLesa I am not sure I understand this completely. This is what I have so far:

- Table `data_entry_iecmaterial2_target_audience` has `iecmaterial2_id`, linking it to `data_entry_iecmaterial2`, and `organisationtarget_id`, linking it to `data_entry_organisationtarget`. 
- In turn, `data_entry_iecmaterial2` has `activity_form_id`, linking it to `data_entry_activityreportform`.
- Currently, there are two views, `vw_organisationtarget` and `vw_activityreportform`, both of which join the respective tables with the stakeholder info. In other views, we often use one of these views.
- Going by the IDs, we could link `activity_form_id` to `vw_activityreportform`, and `organisationtarget_id` to `vw_organisationtarget`, but this is obviously not correct, since we would be pulling in stakeholder info twice (possibly from different records).

So my question is then, do I link both of these IDs, and do I join with the stakeholder table as well? But via which table? Or am I misunderstanding this completely?

To put it another way, so far I have this:

```sql
create or replace view vw_iecmaterial2 as
  select ...,
    imt.id as iecmaterial2_target_audience_id,
    im.id as iecmaterial2_id
  from data_entry_iecmaterial2_target_audience imt
  left outer join data_entry_iecmaterial2 im
    on im.id = imt.iecmaterial2_id
  left outer join ...
    on im.activity_form_id = ...
  left outer join ...
    on imt.organisationtarget_id = ...;
```

...but what else should I join to? And should I use the aforementioned views at all, or are they not a good fit here?

**Comment by AndreLesa** [comment]
Ah ok - I see where the issue is. 
"Going by the IDs, we could link activity_form_id to vw_activityreportform, and organisationtarget_id to vw_organisationtarget, but this is obviously not correct, since we would be pulling in stakeholder info twice (possibly from different records)." 
No you should actually link data_entry_iecmaterial2_target_audience.organisationtarget_id to data_entry_organisationtarget which only has the target_option which is the charfield that we want to use for visualization.

The gist of it is that we are using the data_entry_organisationtarget(which is just a list of options0 in two places: 
1. Under the stakeholder directory (aka stakeholder profile, aka data_entry_stakeholderdirectory) - To figure out who the organisation targets with their services - in general. This is what you have as vw_organisationtarget and this is working fine.
2. Then again under the IEC material, we are recording who their IEC materials targeted - and the list is based again on data_entry_organisationtarget which is remember, just a list list of options. The IEC Material question is answered quarterly as part of the data_entry_activityreport_form.

From the admin interface:
**The data_entry_organisationtarget which is the list can be found as**: https://nacmis.bluecodeltd.com/admin/data_entry/organisationtarget/
**This list is then used first here**: https://nacmis.bluecodeltd.com/admin/data_entry/stakeholderdirectory/82/change/  
While the

**Comment by AndreLesa** [comment]
Fixed.
